src/renderer/components/VideoPreview/VideoPreviewController.tsx

```diff
@@ -225,6 +249,28 @@ const VideoPreviewControllerBase = (
     );
   }, [currentProject?.mediaFilePath]);
 
+  useEffect(() => {
+    if (currentProject !== null && currentProject?.transcription !== null) {
+      cuts.current = convertTranscriptToCuts(
+        currentProject.transcription,
+        rangeOverride
+      );
+
+      if (cuts.current.length === 0) {
+        return;
+      }
+      const lastCut = cuts.current[cuts.current.length - 1];
```

Comment:
Just thinking of DRY as much as applicable, maybe these two lines can be their own function used in both useEffects? eg: `updateOutputVideoLength()`

Reply:
Done here: [c47c7b6](https://github.com/chloebrett/mlvet/pull/333/commits/c47c7b6658e82dd601349eaa762a40e2848a2878)